### PR TITLE
Improved display for Floating Labels

### PR DESCRIPTION
**Pull request by masoodkhoshgard**
### Description

Resolves #16255

### CODE REVIEW

File: framework/src/Volo.Abp.AspNetCore.Mvc.UI.Bootstrap/TagHelpers/Form/AbpInputTagHelperService.cs [hunk 1]
```diff
@@ -227,10 +227,11 @@ protected virtual void AddPlaceholderAttribute(TagHelperOutput inputTagHelperOut
...
 
-        if (attribute != null)
+        if (attribute != null || TagHelper.FloatingLabel)
         {
-            var placeholderLocalized = _tagHelperLocalizer.GetLocalizedText(attribute.Value, TagHelper.AspFor.ModelExplorer);
-
+            var placeholderLocalized = _tagHelperLocalizer.GetLocalizedText(attribute?.Value ?? string.Empty, TagHelper.AspFor.ModelExplorer);
+            if (string.IsNullOrWhiteSpace(placeholderLocalized))
+                placeholderLocalized = TagHelper.AspFor.Name;
             inputTagHelperOutput.Attributes.Add("placeholder", placeholderLocalized);
```
Comment (salihozkara) [inline framework/src/Volo.Abp.AspNetCore.Mvc.UI.Bootstrap/TagHelpers/Form/AbpInputTagHelperService.cs:233]: Why did we add the placeholder attribute here? When I test it, it does not appear on the UI side.

```cs
<abp-input floating-label="true" label="Test" asp-for="MyProperty" />
```

image

File: framework/src/Volo.Abp.AspNetCore.Mvc.UI.Bootstrap/TagHelpers/Form/AbpInputTagHelperService.cs [hunk 2]
```diff
@@ -227,10 +227,11 @@ protected virtual void AddPlaceholderAttribute(TagHelperOutput inputTagHelperOut
...
         var attribute = TagHelper.AspFor.ModelExplorer.GetAttribute<Placeholder>();
 
-        if (attribute != null)
+        if (attribute != null || TagHelper.FloatingLabel)
         {
-            var placeholderLocalized = _tagHelperLocalizer.GetLocalizedText(attribute.Value, TagHelper.AspFor.ModelExplorer);
-
+            var placeholderLocalized = _tagHelperLocalizer.GetLocalizedText(attribute?.Value ?? string.Empty, TagHelper.AspFor.ModelExplorer);
+            if (string.IsNullOrWhiteSpace(placeholderLocalized))
+                placeholderLocalized = TagHelper.AspFor.Name;
```
Comment (salihozkara) [inline framework/src/Volo.Abp.AspNetCore.Mvc.UI.Bootstrap/TagHelpers/Form/AbpInputTagHelperService.cs]: use curly braces

**Comment by masoodkhoshgard** [comment]
Hello @salihozkara.
I checked and it works fine on the UI side.
Would you please check again?

**Comment by salihozkara** [comment]
> Hello @salihozkara. I checked and it works fine on the UI side. Would you please check again?   

Can you try entering it differently from the label property name?
The point here is to see that the placeholder doesn't change anything.

**Comment by masoodkhoshgard** [comment]
> > Hello @salihozkara. I checked and it works fine on the UI side. Would you please check again?   
> 
> Can you try entering it differently from the label property name? The point here is to see that the placeholder doesn't change anything.

Yes. see this

**Comment by salihozkara** [comment]
@masoodkhoshgard  There is no TestString expression in the UI so we don't need to add a placeholder

image

**Comment by masoodkhoshgard** [comment]
@salihozkara If there is no placeholder, it will not work properly. And the label does not move as a float. You can refer to the Bootstrap documentation for more information
https://getbootstrap.com/docs/5.0/forms/floating-labels/

**Comment by salihozkara** [comment]
> @salihozkara If there is no placeholder, it will not work properly. And the label does not move as a float. You can refer to the Bootstrap documentation for more information https://getbootstrap.com/docs/5.0/forms/floating-labels/

Sorry. I didn't realize it wasn't working properly. So should it be a property name or a label value?

**Comment by masoodkhoshgard** [comment]
@salihozkara It doesn't matter what the value of the placeholder is, only its presence is important, and JavaScript replaces the value of the label with the value of the placeholder.

**Review by salihozkara — APPROVED** [review]

**Review by salihozkara — APPROVED** [review]

**Pull request merged**